app/models/web_document.rb

```diff
@@ -19,7 +19,12 @@ def linkdata
   end
 
   def language
-    @language ||= (Language.iso_639_1(extract_language) || detect_language)
```

Comment:
You're setting 'en' as a default value if `Language.iso_639_1` and `detect_language` fail to do it's job. This can be accomplish in here directly, without any extra changes. What do you think?
```ruby
@language ||= Language.iso_639_1(extract_language) || detect_language || 'en'
```